rust/src/base.rs

```diff
@@ -28,6 +28,26 @@ pub struct Base {
     station_id: Arc<Option<StationId>>,
 }
 
+impl Base {
```

Comment:
another lofty thought comment, but, will we want to reproduce the mixin (guessing a rust _trait_) of an Entity supertype here? that was done for Vehicle, Request, Base, and Station, that they extended an abstract base class Entity in our Python code.

Reply:
yeah that's a good point and I think that we would want to implement a trait for an Entity (and I think it would be much cleaner than our frozen dataclass mixin spaghettiish workaround since traits are first class)